pandas/_libs/parsers.pyx

```diff
@@ -1229,7 +1238,20 @@ cdef class TextReader:
                 na_count = 0
 
             if result is not None and dtype != 'int64':
-                result = result.astype(dtype)
+                if is_extension_array_dtype(dtype):
```

Comment:
this ``if is_extension_array_dtype`` should be right after the ``is_categorical_dtye``, IOW on line 1227.